jenkins/Jenkinsfile-release

```diff
@@ -0,0 +1,164 @@
+                        mkdir -p "${env.SPARK_HOME}"
+                        tar zxvf spark-${env.SPARK_VERSION}-bin-hadoop2.7.tgz -C "${env.SPARK_HOME}" --strip-components 1
+                        rm -rf spark-${env.SPARK_VERSION}-bin-hadoop2.7.tgz
+                    """
+            }
+        }
+
+
+        stage('Clone the repository'){
+            steps{
+                    checkout scm
+                }
+        }
+
+        stage('Build Sparkling Water'){
+            steps {
+                withCredentials([file(credentialsId: 'release-gradle.properties', variable: 'GRADLE_PROPERTIES_PATH'),
+                                 file(credentialsId: 'release-secret-key-ring-file', variable: 'RING_FILE_PATH')]) {
+                        sh """
+                            # Update gradle build number
+                            sed -i.backup -E "s/\\.[0-9]+-SNAPSHOT/.$BUILD_NUMBER/" gradle.properties
+                            REL_VERSION=`cat gradle.properties | grep version | grep -v '#' | sed -e "s/.*=//"`
+                            NEXT_BUILD_NUMBER=\$(( $BUILD_NUMBER + 1 ))
+                            NEW_VERSION=`echo "\$REL_VERSION" | sed -E "s/\\.[0-9]+\$/.\${NEXT_BUILD_NUMBER}-SNAPSHOT/"`
+                            echo "org.gradle.jvmargs=-XX:MaxPermSize=512m -Xmx1024m" >> gradle.properties
+                            echo 
+                            echo "gradle.properties"
+                            cat gradle.properties
+                            echo
+                            echo "RELEASE VERSION: \${REL_VERSION}"
+                            echo "NEW VERSION: \${NEW_VERSION}"
+                            echo
+                            
+                            # Fetch H2O python package
+                            mkdir -p private/
+                            curl \$(./gradlew -q printH2OWheelPackage) > private/h2o.whl
+                            export H2O_PYTHON_WHEEL="\$(pwd)/private/h2o.whl"
+                            
+                            # Perform a build
+                            ./gradlew build -x check 
+                            ./gradlew buildSparklingWaterDist
+                            
+                            # Restore gradle.properties before release
+                            git checkout -- gradle.properties
+                            cat ${GRADLE_PROPERTIES_PATH} >> gradle.properties
+                            echo signing.secretKeyRingFile=$RING_FILE_PATH >> gradle.properties
+                            ./gradlew -PdoPromotion -PdoRelease -Pgradle.release.useAutomaticVersion=true -PreleaseVersion="\$REL_VERSION" -PnewVersion="\$NEW_VERSION" release -x check
+                            # Restore gradle.properties before pushing to S3
+                            git checkout -- gradle.properties
+                        """
+                }
+            }
+        }
+
+
+        stage('Create Extended H2O Jars'){
```

Comment:
Tiny detail - a new stage will do  unnecessary checkout from repo.

Reply:
What do you mean please ? We do `checkout` once at the beginning to obtain the code from the repo